storage3/_async/file_api.py

```diff
@@ -144,7 +149,9 @@ async def download(self, path: str) -> bytes:
         """
         _path = self._get_final_path(path)
         response = await self._request(
-            "GET", f"{self._url}/object/{_path}", headers=self._headers
+            "GET",
+            f"{self._client.base_url}object/{_path}",
+            headers=self._client.headers,
```

Comment:
```suggestion
            f"/object/{_path}",
``` 
headers need not be passed again explicitly as you've set it on the httpx Client itself